ircd/send.c

```diff
@@ -1503,3 +1503,169 @@ kill_client_serv_butone(struct Client *one, struct Client *target_p, const char
 
 	rb_linebuf_donebuf(&rb_linebuf_id);
 }
+
+static struct Client *multiline_stashed_target_p;
+static char multiline_prefix[DATALEN+2]; /* allow for trailing space and null termination */
+static int multiline_prefix_len;
+static char multiline_separator[2];
+static int multiline_separator_len;
+static char *multiline_item_start;
+static char *multiline_cur;
+static int multiline_cur_len;
+static int multiline_remote_extra_space;
+
+bool
+sendto_one_multiline_init(struct Client *target_p, const char *separator, const char *format, ...)
+{
+	va_list args;
+
+	s_assert(multiline_stashed_target_p == NULL && "Multiline: didn't cleanup after last usage!");
+
+	va_start(args, format);
+	multiline_prefix_len = vsnprintf(multiline_prefix, sizeof multiline_prefix, format, args);
+	va_end(args);
+
+	if (multiline_prefix_len < 0 || multiline_prefix_len >= DATALEN)
+	{
+		s_assert(false && "Multiline: failure preparing prefix!");
+		return false;
+	}
+
+	multiline_separator_len = rb_strlcpy(multiline_separator, separator, sizeof multiline_separator);
+	if (multiline_separator_len >= sizeof multiline_separator)
+	{
+		s_assert(false && "Multiline: separator too long");
+		return false;
+	}
+
+	multiline_stashed_target_p = target_p;
+	multiline_item_start = multiline_prefix + multiline_prefix_len;
+	multiline_cur = multiline_item_start;
+	multiline_cur_len = multiline_prefix_len;
+	multiline_remote_extra_space = 0;
+	return true;
+}
+
+bool
+sendto_one_multiline_remote_extra_space(struct Client *target_p, struct Client *client_p)
+{
+	ssize_t remote_extra_space;
+
+	if (target_p != multiline_stashed_target_p)
+	{
+		s_assert(false && "Multiline: missed init call!");
+		multiline_stashed_target_p = NULL;
+		return false;
+	}
+
```

Comment:
Why not check MyConnect here and save callers doing it?

```suggestion

	if (!MyConnect(client_p))
		return true;

```